### PR TITLE
bugfix for chomp fixed base joint bug

### PR DESCRIPTION
**Pull request by mlautman**
### Description

This fixes an issue where where the chomp optimizer was requesting the parent joint of a fixed base joint without checking for Null [moveit_tutorials #164](https://github.com/ros-planning/moveit_tutorials/issues/164)

### Checklist
- [x] **Required**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extended the tutorials / documentation, if necessary [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Include a screenshot if changing a GUI
- [ ] Optional: Created tests, which fail without this PR [reference](http://docs.ros.org/kinetic/api/moveit_tutorials/html/doc/tests.html)
- [ ] Optional: Decide if this should be cherry-picked to other current ROS branches (Indigo, Jade, Kinetic)

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: moveit_planners/chomp/chomp_motion_planner/src/chomp_optimizer.cpp [hunk 1]
```diff
@@ -204,7 +204,11 @@ void ChompOptimizer::initialize()
   for (size_t i = 0; i < joint_model_group_->getFixedJointModels().size(); i++)
   {
     const moveit::core::JointModel* model = joint_model_group_->getFixedJointModels()[i];
```
Comment (rhaschke) [inline moveit_planners/chomp/chomp_motion_planner/src/chomp_optimizer.cpp]: Please take the chance and replace the for-loop by `for (const moveit::core::JointModel* model : joint_model_group_->getFixedJointModels())`.

File: moveit_planners/chomp/chomp_motion_planner/src/chomp_optimizer.cpp [hunk 2]
```diff
@@ -204,7 +204,11 @@ void ChompOptimizer::initialize()
...
-    fixed_link_resolution_map[model->getName()] = model->getParentLinkModel()->getParentJointModel()->getName();
+    // In case this is the base joint (eg: a fixed virtual_joint) then we need to check for NULL
+    if (model->getParentLinkModel())
```
Comment (rhaschke) [inline moveit_planners/chomp/chomp_motion_planner/src/chomp_optimizer.cpp]: I would prefer a 
```
if (!model->getParentLinkModel())
  continue;
```
However, that's probably a matter of taste.

Comment (bmagyar) [reply moveit_planners/chomp/chomp_motion_planner/src/chomp_optimizer.cpp]: I think it's better to avoid goto-derivatives in general.

Comment (rhaschke) [reply moveit_planners/chomp/chomp_motion_planner/src/chomp_optimizer.cpp]: As I said, that's a matter of taste. Leave it like it is.

Comment (davetcoleman) [reply moveit_planners/chomp/chomp_motion_planner/src/chomp_optimizer.cpp]: +1 to using ``continue`` or ``return`` early

**Pull request merged**